# TEMPL: A template-based protein ligand pose prediction baseline

J. Fülöp\[1\], M. Šícho\[1\], W. Dehaen\[1\]\[2\]   
\[1\] CZ-OPENSCREEN, Department of Informatics and Chemistry, Faculty of Chemical Technology, University of Chemistry and Technology Prague, Technická 5, 16 628 Prague 6, Czech Republic;  
\[2\] Department of Organic Chemistry, Faculty of Chemical Technology, University of Chemistry and Technology Prague, Technická 5, 16 628 Prague 6, Czech Republic;

# Abstract:

Pose prediction of ligands to proteins remains a central challenge of structure based drug design. Data-driven methods for pose prediction now routinely outperform traditional techniques such as molecular docking, although data leakage and generalizability concerns remain. We propose a simple ligand based baseline for pose prediction which is based on maximal common substructure (MCS) to reference molecules, followed by constrained embedding. As this TEMplate-based Protein Ligand (TEMPL) baseline is strictly data-driven, it is a particularly meaningful baseline for interpolative tasks, where physics based methods sometimes underperform as they exploit data less directly. We applied our baseline method in the ASAP-Polaris-OpenADMET antiviral competition, achieving a result that outperformed some classic docking algorithms for the pose prediction of a series of ligands at the Main Protease of SARS-CoV-2 and MERS-CoV. Furthermore, we show the performance of our baseline is \[relatively good\] on a protein-ligand pose prediction benchmark used for deep learning based pose prediction, PDBBind, highlighting the risk of data leakage and necessity of challenging splits. We also show our baseline method has limited performance for more challenging benchmarks, such as PoseBusters. We provide our baseline method as open source software. For convenience and for non-technical users, we also provide a web application to run the pipeline.

# Introduction:

The prediction of the pose (conformation and absolute placement) of an unseen ligand to a biomacromolecule is one of the grand challenges of structure-based drug design.[1](https://www.zotero.org/google-docs/?gWsK8I)mo The capacity to predict the correct pose of a ligand enables rational design of analogs, as well as providing the starting point for molecular modelling such as molecular dynamics or free energy of binding estimation methods. The traditional method used for pose prediction is molecular docking, which is widely used and has achieved many successes as an important component of drug design campaigns[2](https://www.zotero.org/google-docs/?Sj4lkk), but which also has several well known serious limitations[3,4](https://www.zotero.org/google-docs/?0RTtY9). Molecular docking is based on the combination of a scoring function which ranks a given pose and a search method, which is used to optimize starting poses into the lowest possible score, usually using a heuristic approach.[5](https://www.zotero.org/google-docs/?1trauM)  
Recently, several new methods for pose prediction that are distinct from classic, physics-inspired molecular docking[6,7](https://www.zotero.org/google-docs/?x4cT7s) were developed, which can be broadly divided into two main categories: deep learning based pose prediction and cofolding. In deep learning based pose prediction, large protein-ligand complex structure datasets are leveraged to predict ligand poses using geometric deep learning. Some notable methods are EquiBind[8](https://www.zotero.org/google-docs/?8yG1tV), which is based on E(3)-equivariant geometric deep learning, and DiffDock[9](https://www.zotero.org/google-docs/?RCcslX), a diffusion based approach. Conversely, in cofolding, protein 3D structure prediction and ligand placement are done concurrently, meaning the input is the sequence of the target protein and not a pre-existing 3D structure. The best known example of the cofolding approach is AlphaFold3[10](https://www.zotero.org/google-docs/?AX2cUj) as well as closely related approaches such as Chai[11](https://www.zotero.org/google-docs/?jGaVc3), Boltz[12](https://www.zotero.org/google-docs/?XroFi2), Protenix[13](https://www.zotero.org/google-docs/?oTWvQK), NeuralPlexer[14](https://www.zotero.org/google-docs/?1DdqM7) as well as RosettaFold AllAtom[15](https://www.zotero.org/google-docs/?fujhGh). These methods have all achieved superior performance in pose prediction success on common benchmarks.  
A typical problem of machine learning approaches is limited extrapolative capacity. Analysis showed that the good reported performance of deep learning based pose prediction and cofolding can be partially attributed to similarities between training and test sets. For example, when using the commonly utilized time split approach on the PDBbind set\[cite\], the test set contains identical and very similar proteins and ligands. These issues as well as ways to deal with them have been discussed at length[16–19](https://www.zotero.org/google-docs/?vlTR8J). Additionally, diffusion based ligand placement methods suffer from some unusual issues, such as alteration of the ligand (e.g. inversion of stereocenters, change of bond orders), which is uncommon with physics inspired methods such as docking.  
Recently, cofolding methods achieved a first serious success in a prospective challenge: the pose prediction component of the ASAP-Polaris-OpenADMET antiviral competition, hereafter abbreviated to “The Polaris competition”[20](https://www.zotero.org/google-docs/?M0otZ7). The Polaris competition involved the prediction of the structure of circa 200 protein ligand complexes of inhibitors bound to the SARS-CoV-2 and MERS-CoV Main Protease (MPro). As these true poses were only made available after the challenge concluded, this can be considered an honest benchmark challenge to compare pose prediction methods. In this challenge, cofolding methods outperformed all other methods, particularly traditional pose prediction methods such as FRED[21](https://www.zotero.org/google-docs/?u3d7lK), GLIDE[22](https://www.zotero.org/google-docs/?FRTMAs) and Vina[23](https://www.zotero.org/google-docs/?5Yh7kJ), confirming the reported good performance of cofolding methods cannot solely be attributed to data leakage and similarity.

This success can be partially attributed to the fact a wealth of data exists for the SARS-CoV-2 Main Protease, which is a widely targeted protein[24](https://www.zotero.org/google-docs/?V8VyUP), and which is the target of Nirmaltrevir, Pfizer’s FDA-approved SARS-CoV-2 MPro inhibitor[25](https://www.zotero.org/google-docs/?IOmHvv)N which is clinically used to treat COVID-19 infections. For MERS-CoV a much smaller amount of protein-ligand structures are available, although the structure is relatively similar to that of SARS-CoV-2 MPro. The abundance of available structures meant known ligands could be used as a sort of template or reference to predict the pose of new ligands, with the assumption common parts of different binders will more or less bind in the same location. This same principle is also applied in template-based docking, which is available in FRED, GLIDE and other docking approaches[26–28](https://www.zotero.org/google-docs/?jfQKiV).

To further investigate and isolate the power of the template effect, we designed a baseline method which is based completely on ligand alignment, unlike template-based docking methods. The procedure is based on first finding the best template using Maximal Common Substructure (MCS) algorithms, then generating conformations based on the template, using constrained embedding, and then finally ranking conformation by 3D alignment with the template. This workflow is described in FigXX.

![][image1]  
FigXX.   
Workflow of TEMPL. A. Alignment of the input protein to a set of reference protein ligand complexes can be used to generate the reference ligands to be used as templates. B. Template based, ligand-only strategy for placement prediction. \[TODO: add a part C with an example molecule from Polaris that first gets MCS, then constrained embedding, then reranking\]

# Methods:

The core methods of TEMPL are all internal functions of the RDKit[29](https://www.zotero.org/google-docs/?eevzFh), a cheminformatics library. The procedure used in TEMPL is based on detection of the Maximal Common Substructure (MCS) between the ligand and a set of reference ligands. Then, the best MCS match is used for constrained embedding, generating conformers where the matching atoms from the MCS are locked in the reference coordinates. Then, the generated conformers are ranked using shape or feature alignment to the reference molecule.  
MCS detection was performed using the RascalMCES[30](https://www.zotero.org/google-docs/?GbUwzs) algorithm. This method was preferred over the standard MCS method in RDKit (rdFMCS[31](https://www.zotero.org/google-docs/?fomHMy)) because of its higher speed, which is an important bottleneck when there is a high amount of probe and reference ligands. This method is different from rdFMCS and leverages matching bonds/edges, and as such is actually based on maximum common edge substructure.  
Constrained embedding was performed using the ETKDGv3 method[32](https://www.zotero.org/google-docs/?IEC7VC), which is a commonly used method for conformer generation \[link to a few papers where they say its the best\]. These generated conformers can be optimized using the MMFF94x[33](https://www.zotero.org/google-docs/?42qbBV) or UFF[34](https://www.zotero.org/google-docs/?iZhRfV)(in case of atom types that do not have parameters in MMFF94x) forcefield, keeping the MCS match atoms locked in place.  
Finally, ranking of the conformers resulting from constrained embedding was done using RDKit’s Align3D method. This recently added method is based on the Gaussian volume approximation of molecular shape which allows fast calculation of overlap[35](https://www.zotero.org/google-docs/?45UZ4g). This overlap is quantified using Shape and ColorTanimoto, which corresponds to the intersection of the (colored or not) volumes divided by the union of the volumes. Another ranking method is ComboTanimoto, which is simply the average between ShapeTanimoto and ColorTanimoto.

*Data*  
Data related to the Polaris competition was obtained using the Polaris Python tool’s data loader. The competition organizers provided reference complexes for the SARS-CoV-2 and MERS-CoV MPro. These were used as references to align all provided experimental SARS-CoV-2 structures to, using the Biotite Python package[36](https://www.zotero.org/google-docs/?x5e038). No MERS-CoV-2 complexes were provided, but a selection of 17 were extracted from the literature. SARS-CoV-2 MPro ligands were then aligned to the reference MERS-CoV MPro complex to augment the data, as these ligands are chemically much closer to the test set molecules, which both for MERS-CoV and SARS-CoV-2 MPro mostly belong to the same series of isoquinoline-based inhibitors. Finally, the superposed ligands were inspected and anything far from the pocket (e.g. bound to the wrong protein chain) was manually removed. This then led to a set of absolute ligand coordinates which could be used as inputs in the ligand-based core TEMPL pipeline.

For the PDBbind experiments, significant data cleaning was performed on the PDBBind[37](https://www.zotero.org/google-docs/?7HrKM8) (v2020) data set. To ensure correct parsing of molecules, which are provided as SDF files, the corresponding ligand SMILES were retrieved \[link to the paper from which you got those SMILES or the repo in case there is no paper\] and used to reconstruct and correct the atom connectivity, which is essential for correct detection of MCS. Nonetheless, X molecules were rejected due to problems with parsing. In the end, this resulted in a training set size of X complexes, a test set size of Y complexes and a validation set size of Z complexes.

Posebusters data was used as provided[38](https://www.zotero.org/google-docs/?abOofT).

The sequences of PDBbind protein chains were embedded using ESM2[39](https://www.zotero.org/google-docs/?XncvFn) embeddings (specifically the esm2\_t33\_650M\_UR50D model), averaging the per residue embeddings into a single 1280 float embedding vector for each input protein sequence. The distance between different sequences were quantified using cosine distance, with the best 100 template proteins retained, and additionally a distance threshold of xx applied to exclude protein sequences with too high dissimilarity. In the case of some longer sequences, it is possible to reduce high GPU memory demands using chunking, the effect of which is shown in SI fig X. To increase portability of our method, we have also made the used embedding available via a publicly available repository to spare the user the effort of having to recalculate these for all proteins in the PDBBind set.

For protein backbone superposition, we applied the “superimpose homologs” method from Biotite. When the final CAlpha RMSD was above a specified threshold (default of X angstrom), the protein was discarded due to low overlap.

For calculation of ligand pose Root Mean Square Deviations (RMSDs), the sPyRMSD Python package[40](https://www.zotero.org/google-docs/?N1xGSh) was used. This method is symmetry-corrected and is also independent of hydrogens and bond orders, which enables some tautomer and protomer related non-identity issues to be averted. The equation for RMSD is given below, with A and B the N by 3 matrices of atomic coordinates of two conformers A and B with matched heavy atoms.  
RMSD \= 1N1N02(Aij-Bij)2  
Generally, an RMSD of 2 Angstrom and below has been considered a docking success although this threshold is considered crude and a truly successful docking prediction will also involve other factors such as physical soundness, retrieval of molecular interactions and realistic torsion angles[38](https://www.zotero.org/google-docs/?shNly3).

Our method is offered as a Python-based command-line tool, and additionally, the tool is made available as a web application. This web application is based on Streamlit and makes it possible for users to try out the TEMPL pipeline without commitment, local installation or familiarity with command-line software. The web application can be run locally, but we have made an instance of it available via [https://templ.dyn.cloud.e-infra.cz](https://templ.dyn.cloud.e-infra.cz) . Instruction for a basic usage task for both the command line tool and the web app are given in the **Supporting Information**.

# Results:

## Polaris competition pose prediction

A prototype version of our method was applied in the pose prediction component of the Polaris competition. This challenge involved predicting the pose of a series of circa 200 closely related ligands to the SARS-CoV-2 and MERS-CoV MPro. A training set of about 800 complexes to SARS-CoV-2 Main protease was made available as a part of the challenge. The initial prototype of our method achieved a success rate of 50%\[factcheck this number\] on the intermediary leaderboard. On the final leaderboard, the performance of the baseline deteriorated to 28%\[factcheck this number\], mainly due to using a smaller template ligand set for the MERS-CoV-2 Main protease and not using SARS-CoV-2 Main protease data. Below, we report the results for the finalized version of TEMPL. We performed an ablation study, of which the results are summarized in Table X. The final maximally achieved performance reported in this paper is 72.5%\[Factcheck this number\]. The \~20% increase in performance versus the leaderboard performance can be attributed to enhanced data cleanup, polishing of the prototype, and the capacity running the ablation study on unblinded test data. 

Table X showing “ablation study” results for:

| Settings | \< 2 A (Test set) | \< 5 A (Test set) | Average RMSD |
| :---- | :---- | :---- | :---- |
| **MCS+200 confs+ComboTanimoto** | **\#\#** | **\#\#** | **\#\#** |
| MCS+200 confs+ShapeTanimoto |  |  |  |
| MCS+200 confs+ColorTanimoto |  |  |  |
| No MCS+200 confs+ComboTanimoto |  |  |  |
| No MCS+200 confs+ShapeTanimoto |  |  |  |
| No MCS+200 confs+ColorTanimoto |  |  |  |
| MCS+200 confs+MMFF94x+ComboTanimoto |  |  |  |
| MCS+200 confs+MMFF94x+ShapeTanimoto |  |  |  |
| MCS+200 confs+MMFF94x+ColorTanimoto |  |  |  |
| MCS+100 confs+ComboTanimoto |  |  |  |
| MCS+100 confs+ShapeTanimoto |  |  |  |
| MCS+100 confs+ColorTanimoto |  |  |  |
| MCS+50 confs+ComboTanimoto |  |  |  |
| MCS+50 confs+ShapeTanimoto |  |  |  |
| MCS+50 confs+ColorTanimoto |  |  |  |
| MCS+20 confs+ComboTanimoto |  |  |  |
| MCS+20 confs+ShapeTanimoto |  |  |  |
| MCS+20 confs+ColorTanimoto |  |  |  |
| MCS+10 confs+ComboTanimoto |  |  |  |
| MCS+10 confs+ShapeTanimoto |  |  |  |
| MCS+10 confs+ColorTanimoto |  |  |  |
| MCS \+ 1 conf |  |  |  |

Table X: 

Table X showing set per set results for MERS and SARS and reference 17 vs transposed SARSes

Figure comparing the TEMPL results with winner and best physics best; for MERS, SARS and combined (using 3x 3 bars); probably just for RMSD \< 2

## PDBbind pose prediction

To compare our baseline head to head with recent pose prediction methods, we employed the PDBbind data set, using commonly used splits, in particular the time split used by Staerk and coworkers[8,9](https://www.zotero.org/google-docs/?8oIdqv). The PDBbind data set is a curated subset of the Protein Data Bank (PDB) with protein-ligand complexes, which consists of XX,XXX unique structures. This split was later criticized for test-train similarities, both on the ligand and protein level, making the pose prediction task easier than anticipated and possibly inflating the performance of data driven pose prediction methods. Because our baseline is designed to capture such effects, we anticipated this is one place where our baseline method should show some degree of successful pose predictions due to near-neighbor behavior.

Using these provided splits, which is based on a cutoff at a given date (1st of january 2020), and using the PDBBind train set complexes as a reference, TEMPL achieves an RMSD success rate of xx%, comparable with the reported success rate of Vina and GLIDE at yy% and xx%. Equibind and DiffDock itself achieved a success rate of vv%. In FigX, the success rate dependency on training set similarity is shown both for ligand and protein similarity. As expected, particularly in the situation of high ligand similarity combined with high protein similarity, high success rates (ww%) are achieved.  In the case of the reported numbers for the training set, we report numbers as a Leave-One-Out (LOO) experiment.

Optimization of *k*, the amount of nearest neighbor proteins from the reference set (the PDBbind training fold) was investigated. Table XX shows that k=xxx is the optimum balance between completeness and performance.

Figure X: heatmap with on one axis: input ligand tanimoto to nearest neighbor in reference set, on other axis: input protein sequence ESM cosine distance to nearest neighbor in reference set and RMSD \< 2 success % in each bin. CAn possibly put a ligand only and protein only similarity curve on the axes, depending on how good that looks

## Posebusters pose prediction and pose evaluation

One of the first serious challenges to the reported good performance of deep learning based pose placement methods was made by the PoseBusters team, which showed these methods suffer from serious problems including physically improbable conformations and steric clashes. The PoseBusters authors also provided a more challenging protein ligand complex data set which was later used as a benchmark set for cofolding methods such as alphafold3, chai, protenix, boltz1, etc. Using the same workflow, TEMPL achieved a success rate of xx%, comparable with the reported success rate of yy% of Vina, and lower than the success rate of xx% reported for AlphaFold3.

Fig.X: show posebusters with best PB, AlphaFold, Equibind and DiffDock, stratified by similarity \[though we don’t have this for AF3 ?\]

Additionally, we used PoseBusters to estimate the PoseBusters validity of poses obtained using TEMPL. As TEMPL is based on MCS alignment and is strictly ligand-based, there is no awareness at all of possible steric clashes, and the aim is solely to obtain poses that perform well on the RMSD based metric. As expected, xx% of the obtained poses were PoseBusters invalid, mainly due to steric clashes. This is comparable with the yy% obtained for DiffDock and EquiBind, but significantly higher than the low invalidity reported for AF3.

# Discussion:

The pose prediction experiments described in the results section of this paper confirm the initial observation from our submission to the Polaris competition. When similar protein-ligand complexes are available, pose prediction based on similarity achieves high success rates. As described elsewhere[16–18](https://www.zotero.org/google-docs/?N8wSPj) the same effect leads to overestimation of the strength of cofolding and deep learning based pose prediction methods.   
The Polaris competition itself corresponded to a “best-case scenario” for similarity based methods: a large amount of Protein-Ligand complex structures were available for SARS-CoV-2 Main Protease and the ligand in the test set mostly belonged to one series of similar compounds contain a decorated isoquinoline. In this case, our method performed well and outperformed traditional methods, including AutoDock Vina with template placement correction. On the other hand, fine-tuned cofolding methods performed significantly better, occupying the entire top 5 in the final leaderboard.  
Applying this method on widely used pose prediction benchmarks confirmed a strong dependency on similarity and achieved average performance, comparable with reported numbers for \[insert vina or glide here\]. As shown in FigX, on novel proteins and/or using novel ligands, the performance completely collapses, confirming this approach is, unlike traditional docking methods\[cite\], not able to extrapolate meaningfully at all.

The ablation study had the surprising result that forcefield optimization of poses did not increase the success rate. It is known ETKDG generated conformers create very high quality ensembles of conformations[41](https://www.zotero.org/google-docs/?x17qUl), but force field optimization is not expected to make the performance worse, or to lower conformation diversity. We attribute this performance degradation to the constraints applied during minimization: because MCS-matched atoms are kept locked, force field minimization is forced to occur in an unusual way. Additionally, the ablation study showed that 3D alignment on its own is able to capture some correct poses too (X%), although it underperforms relative to our optimized baseline. The ablation study showed all 3 components, MCS, constrained embedding and alignment based placement ranking are necessary for performance.

Surprisingly, the PoseBusters validity of our outputs was relatively high. This is interesting, because all information about possible steric clashes or protein contacts is only indirectly provided by the shape of the reference ligands. Nonetheless, this is enough to prevent the majority of steric clashes. Because conformations originate from ETKDG, only a limited amount of physically unrealistic conformers occur.

*Limitations:*  
TEMPL is a baseline method and is expected to underperform in challenging tasks. This is seen in the PoseBusters protein-ligand set, where we observed a low recovery rate of correct poses. Comparably with what is reported for EquiBind in the PoseBusters pose prediction task, we observe a distinct relation between pose recovery success and the similarity of protein sequences to the training set. Notably, modern cofolding methods such as AlphaFold3 perform well on this task.

# Conclusion:

We have constructed a baseline method which leverages knowledge from known protein ligand complexes to predict the poses of new ligands to new proteins. We show this baseline achieves relatively strong performance at pose prediction, sometimes outperforming widely used traditional pose prediction methods such as AutoDock Vina. Although our baseline was not among the highest ranked in the final Polaris competition leaderboard, the final optimized baseline outperforms many advanced pose prediction techniques and matches the performance of the best physics-based methods. We propose our baseline to be used as a realistic minimal baseline that needs to be outperformed by newly proposed pose prediction methods, particularly under conditions where highly similar protein or ligand information can be leveraged. Our method is available both as open-source code as well as through a web app, accessible at [https://templ.dyn.cloud.e-infra.cz/](https://templ.dyn.cloud.e-infra.cz/) . 

## Acknowledgement

J.F., M.Š. and W.D. were supported by the Ministry of Education, Youth and Sports of the Czech Republic – National Infrastructure for Chemical Biology (CZ-OPENSCREEN, LM2023052). W.D. was supported by the Ministry of Education, Youth and Sports of the Czech Republic by the project “New Technologies for Translational Research in Pharmaceutical Sciences/NETPHARM”, project ID CZ.02.01.01/00/22\_008/0004607, cofunded by the European Union. 

## Supplementary materials

Code examples for basic usage are available in the Supporting Information.  
SI.docx (should contain some of the plots from ESM like the TSNE and maybe the chunking info and whatever else is irrelevant to the main line of the paper or whatever doesnt fit in the space constraints). This we cann drop on the bottom of this document and ill export it as a separate file later.

## Code and data availability

The source code of TEMPL is freely available via [github.com/fulopjoz/templ-pipeline](http://github.com/fulopjoz/templ-pipeline), and is released under the permissive MIT license. This also includes the code for a streamlit application, which can be run locally or accessed via [https://templ.dyn.cloud.e-infra.cz/](https://templ.dyn.cloud.e-infra.cz/) . Data is deposited to Zenodo including a snapshot of the code at the time of submission and precalculated ESM embeddings. 

# References

[(1)	Gschwend, D. A.; Good, A. C.; Kuntz, I. D. Molecular Docking towards Drug Discovery. *J. Mol. Recognit.* **1996**, *9* (2), 175–186. https://doi.org/10.1002/(SICI)1099-1352(199603)9:2\<175::AID-JMR260\>3.0.CO;2-D.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(2)	Sabe, V. T.; Ntombela, T.; Jhamba, L. A.; Maguire, G. E. M.; Govender, T.; Naicker, T.; Kruger, H. G. Current Trends in Computer Aided Drug Design and a Highlight of Drugs Discovered via Computational Techniques: A Review. *European Journal of Medicinal Chemistry* **2021**, *224*, 113705\. https://doi.org/10.1016/j.ejmech.2021.113705.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(3)	Gentile, F.; Oprea, T. I.; Tropsha, A.; Cherkasov, A. Surely You Are Joking, Mr Docking\! *Chem. Soc. Rev.* **2023**, *52* (3), 872–878. https://doi.org/10.1039/D2CS00948J.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(4)	Warren, G. L.; Andrews, C. W.; Capelli, A.-M.; Clarke, B.; LaLonde, J.; Lambert, M. H.; Lindvall, M.; Nevins, N.; Semus, S. F.; Senger, S.; Tedesco, G.; Wall, I. D.; Woolven, J. M.; Peishoff, C. E.; Head, M. S. A Critical Assessment of Docking Programs and Scoring Functions. *J. Med. Chem.* **2006**, *49* (20), 5912–5931. https://doi.org/10.1021/jm050362n.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(5)	Dias, R.; De Azevedo Jr., W. Molecular Docking Algorithms. *CDT* **2008**, *9* (12), 1040–1047. https://doi.org/10.2174/138945008786949432.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(6)	Pagadala, N. S.; Syed, K.; Tuszynski, J. Software for Molecular Docking: A Review. *Biophys Rev* **2017**, *9* (2), 91–102. https://doi.org/10.1007/s12551-016-0247-1.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(7)	Onodera, K.; Satou, K.; Hirota, H. Evaluations of Molecular Docking Programs for Virtual Screening. *J. Chem. Inf. Model.* **2007**, *47* (4), 1609–1618. https://doi.org/10.1021/ci7000378.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(8)	Stärk, H.; Ganea, O.-E.; Pattanaik, L.; Barzilay, R.; Jaakkola, T. EquiBind: Geometric Deep Learning for Drug Binding Structure Prediction. arXiv June 4, 2022\. https://doi.org/10.48550/arXiv.2202.05146.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(9)	Corso, G.; Stärk, H.; Jing, B.; Barzilay, R.; Jaakkola, T. DiffDock: Diffusion Steps, Twists, and Turns for Molecular Docking. arXiv February 11, 2023\. https://doi.org/10.48550/arXiv.2210.01776.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(10)	Abramson, J.; Adler, J.; Dunger, J.; Evans, R.; Green, T.; Pritzel, A.; Ronneberger, O.; Willmore, L.; Ballard, A. J.; Bambrick, J.; Bodenstein, S. W.; Evans, D. A.; Hung, C.-C.; O’Neill, M.; Reiman, D.; Tunyasuvunakool, K.; Wu, Z.; Žemgulytė, A.; Arvaniti, E.; Beattie, C.; Bertolli, O.; Bridgland, A.; Cherepanov, A.; Congreve, M.; Cowen-Rivers, A. I.; Cowie, A.; Figurnov, M.; Fuchs, F. B.; Gladman, H.; Jain, R.; Khan, Y. A.; Low, C. M. R.; Perlin, K.; Potapenko, A.; Savy, P.; Singh, S.; Stecula, A.; Thillaisundaram, A.; Tong, C.; Yakneen, S.; Zhong, E. D.; Zielinski, M.; Žídek, A.; Bapst, V.; Kohli, P.; Jaderberg, M.; Hassabis, D.; Jumper, J. M. Accurate Structure Prediction of Biomolecular Interactions with AlphaFold 3\. *Nature* **2024**, *630* (8016), 493–500. https://doi.org/10.1038/s41586-024-07487-w.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(11)	Chai Discovery; Boitreaud, J.; Dent, J.; McPartlon, M.; Meier, J.; Reis, V.; Rogozhnikov, A.; Wu, K. Chai-1: Decoding the Molecular Interactions of Life. Synthetic Biology October 11, 2024\. https://doi.org/10.1101/2024.10.10.615955.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(12)	Wohlwend, J.; Corso, G.; Passaro, S.; Getz, N.; Reveiz, M.; Leidal, K.; Swiderski, W.; Atkinson, L.; Portnoi, T.; Chinn, I.; Silterra, J.; Jaakkola, T.; Barzilay, R. Boltz-1 Democratizing Biomolecular Interaction Modeling. Biophysics November 20, 2024\. https://doi.org/10.1101/2024.11.19.624167.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(13)	ByteDance AML AI4Science Team; Chen, X.; Zhang, Y.; Lu, C.; Ma, W.; Guan, J.; Gong, C.; Yang, J.; Zhang, H.; Zhang, K.; Wu, S.; Zhou, K.; Yang, Y.; Liu, Z.; Wang, L.; Shi, B.; Shi, S.; Xiao, W. Protenix \- Advancing Structure Prediction Through a Comprehensive AlphaFold3 Reproduction. Bioinformatics January 11, 2025\. https://doi.org/10.1101/2025.01.08.631967.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(14)	Qiao, Z.; Nie, W.; Vahdat, A.; Miller, T. F.; Anandkumar, A. State-Specific Protein–Ligand Complex Structure Prediction with a Multiscale Deep Generative Model. *Nat Mach Intell* **2024**, *6* (2), 195–208. https://doi.org/10.1038/s42256-024-00792-z.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(15)	Krishna, R.; Wang, J.; Ahern, W.; Sturmfels, P.; Venkatesh, P.; Kalvet, I.; Lee, G. R.; Morey-Burrows, F. S.; Anishchenko, I.; Humphreys, I. R.; McHugh, R.; Vafeados, D.; Li, X.; Sutherland, G. A.; Hitchcock, A.; Hunter, C. N.; Kang, A.; Brackenbrough, E.; Bera, A. K.; Baek, M.; DiMaio, F.; Baker, D. Generalized Biomolecular Modeling and Design with RoseTTAFold All-Atom. *Science* **2024**, *384* (6693), eadl2528. https://doi.org/10.1126/science.adl2528.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(16)	Jain, A. N.; Cleves, A. E.; Walters, W. P. Deep-Learning Based Docking Methods: Fair Comparisons to Conventional Docking Workflows. arXiv 2024\. https://doi.org/10.48550/ARXIV.2412.02889.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(17)	Masters, M. R.; Mahmoud, A. H.; Lill, M. A. Do Deep Learning Models for Co-Folding Learn the Physics of Protein-Ligand Interactions? Bioinformatics June 4, 2024\. https://doi.org/10.1101/2024.06.03.597219.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(18)	Yu, Y.; Lu, S.; Gao, Z.; Zheng, H.; Ke, G. Do Deep Learning Models Really Outperform Traditional Approaches in Molecular Docking? arXiv 2023\. https://doi.org/10.48550/ARXIV.2302.07134.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(19)	Škrinjar, P.; Eberhardt, J.; Durairaj, J.; Schwede, T. Have Protein-Ligand Co-Folding Methods Moved beyond Memorisation? Bioinformatics February 7, 2025\. https://doi.org/10.1101/2025.02.03.636309.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(20)	MacDermott-Opeskin, H.; Scheen, J.; Wognum, C.; T Horton, J.; West, D.; Payne, A. M.; Castellanos, M. A.; Colby, S.; Griffen, E.; Cousins, D.; Stacey, J.; Reid, L.; Cara Aschenbrenner, J.; Fearon, D.; Balcomb, B.; Marples, P.; W.E. Tomlinson, C.; Lithgo, R.; Winokan, M.; Barr, H.; Lahav, N.; Lavi, M.; Duberstein, S.; Cohen, G.; Fate, G.; Lefker, B.; Robinson, R.; Szommer, T.; Lynch, N.; Tollefson, M.; Xu, C.; Hsu, J.; St-Laurent, J.; Etsmoberg, H.; Zhu, L.; Quirke, A.; Abdul Haleem, M. I.; Alibay, I.; Baid, G.; Birnbaum, B.; Bishop, K.; Bohorquez, H.; Bose, A.; Brown, C. J.; Burns, J.; Cai, L.; Cedeno, R.; Chupakhin, V.; Clark, F.; Cole, D.; Corbi-Verge, C.; Danial, M.; Davi, A.; Dehaen, W.; Doering, N. P.; Dougha, A.; Eakin, B.; Ehrlich, A.; Elijosius, R.; Fülöp, J.; Gitter, A.; Gu, Y.; Head-Gordon, T.; Jiang, E.; Kaminow, B.; Khosravi, S.; Khoualdi, A. F.; Lenselink, E. B.; Liu, Z.; Liu, Y.; Liu, S.; Ma, Y.; Maher, P.; Mayer, I.; Mey, A.; Montanari, F.; Niu, T.; Ogino, R.; Palaniappan, A.; Pan, X.; Patnaik, A.; Pham, L.-H.; Pinto, L.; Purnomo, J.; Rich, A.; Schaaf, L.; Schran, C.; Srivastava, S. P.; Sun, K.; Sun, Z.; Talagayev, V.; Thirukonda Subramanian Balakrishnan, B.; Tkatchenko, A.; Treyde, W.; Tripp, A.; Vithayapalert, N.; Wang, Y.; Wasi, A. T.; Wedig, S.; Xu, B.; Zhou, W.; Von Delft, F.; Lee, A.; Kirkegaard, K.; Sjö, P.; Fraser, J.; Chodera, J. D. A Computational Community Blind Challenge on Pan-Coronavirus Drug Discovery Data. Chemistry July 25, 2025\. https://doi.org/10.26434/chemrxiv-2025-zd9mr.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(21)	McGann, M. FRED Pose Prediction and Virtual Screening Accuracy. *J. Chem. Inf. Model.* **2011**, *51* (3), 578–596. https://doi.org/10.1021/ci100436p.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(22)	Friesner, R. A.; Banks, J. L.; Murphy, R. B.; Halgren, T. A.; Klicic, J. J.; Mainz, D. T.; Repasky, M. P.; Knoll, E. H.; Shelley, M.; Perry, J. K.; Shaw, D. E.; Francis, P.; Shenkin, P. S. Glide: A New Approach for Rapid, Accurate Docking and Scoring. 1\. Method and Assessment of Docking Accuracy. *J. Med. Chem.* **2004**, *47* (7), 1739–1749. https://doi.org/10.1021/jm0306430.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(23)	Trott, O.; Olson, A. J. AutoDock Vina: Improving the Speed and Accuracy of Docking with a New Scoring Function, Efficient Optimization, and Multithreading. *J Comput Chem* **2010**, *31* (2), 455–461. https://doi.org/10.1002/jcc.21334.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(24)	Boby, M. L.; Fearon, D.; Ferla, M.; Filep, M.; Koekemoer, L.; Robinson, M. C.; The COVID Moonshot Consortium‡; Chodera, J. D.; Lee, A. A.; London, N.; Von Delft, A.; Von Delft, F.; Achdout, H.; Aimon, A.; Alonzi, D. S.; Arbon, R.; Aschenbrenner, J. C.; Balcomb, B. H.; Bar-David, E.; Barr, H.; Ben-Shmuel, A.; Bennett, J.; Bilenko, V. A.; Borden, B.; Boulet, P.; Bowman, G. R.; Brewitz, L.; Brun, J.; Bvnbs, S.; Calmiano, M.; Carbery, A.; Carney, D. W.; Cattermole, E.; Chang, E.; Chernyshenko, E.; Clyde, A.; Coffland, J. E.; Cohen, G.; Cole, J. C.; Contini, A.; Cox, L.; Croll, T. I.; Cvitkovic, M.; De Jonghe, S.; Dias, A.; Donckers, K.; Dotson, D. L.; Douangamath, A.; Duberstein, S.; Dudgeon, T.; Dunnett, L. E.; Eastman, P.; Erez, N.; Eyermann, C. J.; Fairhead, M.; Fate, G.; Fedorov, O.; Fernandes, R. S.; Ferrins, L.; Foster, R.; Foster, H.; Fraisse, L.; Gabizon, R.; García-Sastre, A.; Gawriljuk, V. O.; Gehrtz, P.; Gileadi, C.; Giroud, C.; Glass, W. G.; Glen, R. C.; Glinert, I.; Godoy, A. S.; Gorichko, M.; Gorrie-Stone, T.; Griffen, E. J.; Haneef, A.; Hassell Hart, S.; Heer, J.; Henry, M.; Hill, M.; Horrell, S.; Huang, Q. Y. J.; Huliak, V. D.; Hurley, M. F. D.; Israely, T.; Jajack, A.; Jansen, J.; Jnoff, E.; Jochmans, D.; John, T.; Kaminow, B.; Kang, L.; Kantsadi, A. L.; Kenny, P. W.; Kiappes, J. L.; Kinakh, S. O.; Kovar, B.; Krojer, T.; La, V. N. T.; Laghnimi-Hahn, S.; Lefker, B. A.; Levy, H.; Lithgo, R. M.; Logvinenko, I. G.; Lukacik, P.; Macdonald, H. B.; MacLean, E. M.; Makower, L. L.; Malla, T. R.; Marples, P. G.; Matviiuk, T.; McCorkindale, W.; McGovern, B. L.; Melamed, S.; Melnykov, K. P.; Michurin, O.; Miesen, P.; Mikolajek, H.; Milne, B. F.; Minh, D.; Morris, A.; Morris, G. M.; Morwitzer, M. J.; Moustakas, D.; Mowbray, C. E.; Nakamura, A. M.; Neto, J. B.; Neyts, J.; Nguyen, L.; Noske, G. D.; Oleinikovas, V.; Oliva, G.; Overheul, G. J.; Owen, C. D.; Pai, R.; Pan, J.; Paran, N.; Payne, A. M.; Perry, B.; Pingle, M.; Pinjari, J.; Politi, B.; Powell, A.; Pšenák, V.; Pulido, I.; Puni, R.; Rangel, V. L.; Reddi, R. N.; Rees, P.; Reid, S. P.; Reid, L.; Resnick, E.; Ripka, E. G.; Robinson, R. P.; Rodriguez-Guerra, J.; Rosales, R.; Rufa, D. A.; Saar, K.; Saikatendu, K. S.; Salah, E.; Schaller, D.; Scheen, J.; Schiffer, C. A.; Schofield, C. J.; Shafeev, M.; Shaikh, A.; Shaqra, A. M.; Shi, J.; Shurrush, K.; Singh, S.; Sittner, A.; Sjö, P.; Skyner, R.; Smalley, A.; Smeets, B.; Smilova, M. D.; Solmesky, L. J.; Spencer, J.; Strain-Damerell, C.; Swamy, V.; Tamir, H.; Taylor, J. C.; Tennant, R. E.; Thompson, W.; Thompson, A.; Tomásio, S.; Tomlinson, C. W. E.; Tsurupa, I. S.; Tumber, A.; Vakonakis, I.; Van Rij, R. P.; Vangeel, L.; Varghese, F. S.; Vaschetto, M.; Vitner, E. B.; Voelz, V.; Volkamer, A.; Walsh, M. A.; Ward, W.; Weatherall, C.; Weiss, S.; White, K. M.; Wild, C. F.; Witt, K. D.; Wittmann, M.; Wright, N.; Yahalom-Ronen, Y.; Yilmaz, N. K.; Zaidmann, D.; Zhang, I.; Zidane, H.; Zitzmann, N.; Zvornicanin, S. N. Open Science Discovery of Potent Noncovalent SARS-CoV-2 Main Protease Inhibitors. *Science* **2023**, *382* (6671), eabo7201. https://doi.org/10.1126/science.abo7201.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(25)	Tuttle, J. B.; Allais, C.; Allerton, C. M. N.; Anderson, A. S.; Arcari, J. T.; Aschenbrenner, L. M.; Avery, M.; Bellenger, J.; Berritt, S.; Boras, B.; Boscoe, B. P.; Buzon, L. M.; Cardin, R. D.; Carlo, A. A.; Coffman, K. J.; Dantonio, A.; Di, L.; Eng, H.; Farley, K. A.; Ferre, R. A.; Gajiwala, K. S.; Gibson, S. A.; Greasley, S. E.; Hurst, B. L.; Kadar, E. P.; Kalgutkar, A. S.; Lachapelle, E. A.; Lanyon, L. F.; Lee, J.; Lee, J.; Lian, Y.; Liu, W.; Martínez-Alsina, L. A.; Mason, S. W.; Noell, S.; Novak, J.; Obach, R. S.; Ogilvie, K.; O’Neil, S. V.; Ostner, G.; Owen, D. R.; Patel, N. C.; Pettersson, M.; Singh, R. S.; Rai, D. K.; Reese, M. R.; Sakata, S.; Sammons, M. F.; Sathish, J. G.; Sharma, R.; Steppan, C. M.; Stewart, A.; Updyke, L.; Verhoest, P. R.; Wei, L.; Wright, S. W.; Yang, E.; Yang, Q.; Zhu, Y. Discovery of Nirmatrelvir (PF-07321332): A Potent, Orally Active Inhibitor of the Severe Acute Respiratory Syndrome Coronavirus 2 (SARS CoV-2) Main Protease. *J. Med. Chem.* **2025**, *68* (7), 7003–7030. https://doi.org/10.1021/acs.jmedchem.4c02561.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(26)	Alekseenko, A.; Kotelnikov, S.; Ignatov, M.; Egbert, M.; Kholodov, Y.; Vajda, S.; Kozakov, D. ClusPro LigTBM: Automated Template-Based Small Molecule Docking. *Journal of Molecular Biology* **2020**, *432* (11), 3404–3410. https://doi.org/10.1016/j.jmb.2019.12.011.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(27)	Kumar, A.; Zhang, K. Y. J. A Pose Prediction Approach Based on Ligand 3D Shape Similarity. *J Comput Aided Mol Des* **2016**, *30* (6), 457–469. https://doi.org/10.1007/s10822-016-9923-2.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(28)	Yang, X.; Liu, Y.; Gan, J.; Xiao, Z.-X.; Cao, Y. FitDock: Protein–Ligand Docking by Template Fitting. *Briefings in Bioinformatics* **2022**, *23* (3), bbac087. https://doi.org/10.1093/bib/bbac087.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(29)	Greg Landrum; Paolo Tosco; Brian Kelley; Ricardo Rodriguez; David Cosgrove; Riccardo Vianello; sriniker; Peter Gedeck; Gareth Jones; Eisuke Kawashima; NadineSchneider; Dan Nealschneider; Andrew Dalke; tadhurst-cdd; Matt Swain; Brian Cole; Samo Turk; Aleksandr Savelev; Alain Vaucher; Maciej Wójcikowski; Ichiru Take; Hussein Faara; Vincent F. Scalfani; Rachel Walker; Daniel Probst; Kazuya Ujihara; Niels Maeder; Jeremy Monat; Juuso Lehtivarjo; guillaume godin. Rdkit/Rdkit: 2025\_03\_5 (Q1 2025\) Release, 2025\. https://doi.org/10.5281/ZENODO.16439048.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(30)	Raymond, J. W. RASCAL: Calculation of Graph Similarity Using Maximum Common Edge Subgraphs. *The Computer Journal* **2002**, *45* (6), 631–644. https://doi.org/10.1093/comjnl/45.6.631.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(31)	Dalke, A.; Hastings, J. FMCS: A Novel Algorithm for the Multiple MCS Problem. *J Cheminform* **2013**, *5* (S1), O6. https://doi.org/10.1186/1758-2946-5-S1-O6.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(32)	Wang, S.; Witek, J.; Landrum, G. A.; Riniker, S. Improving Conformer Generation for Small Rings and Macrocycles Based on Distance Geometry and Experimental Torsional-Angle Preferences. *J. Chem. Inf. Model.* **2020**, *60* (4), 2044–2058. https://doi.org/10.1021/acs.jcim.0c00025.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(33)	Halgren, T. A. Merck Molecular Force Field. I. Basis, Form, Scope, Parameterization, and Performance of MMFF94. *J. Comput. Chem.* **1996**, *17* (5–6), 490–519. https://doi.org/10.1002/(SICI)1096-987X(199604)17:5/6\<490::AID-JCC1\>3.0.CO;2-P.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(34)	Rappe, A. K.; Casewit, C. J.; Colwell, K. S.; Goddard, W. A.; Skiff, W. M. UFF, a Full Periodic Table Force Field for Molecular Mechanics and Molecular Dynamics Simulations. *J. Am. Chem. Soc.* **1992**, *114* (25), 10024–10035. https://doi.org/10.1021/ja00051a040.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(35)	Grant, J. A.; Gallardo, M. A.; Pickup, B. T. A Fast Method of Molecular Shape Comparison: A Simple Application of a Gaussian Description of Molecular Shape. *J. Comput. Chem.* **1996**, *17* (14), 1653–1666. https://doi.org/10.1002/(SICI)1096-987X(19961115)17:14\<1653::AID-JCC7\>3.0.CO;2-K.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(36)	Kunzmann, P.; Hamacher, K. Biotite: A Unifying Open Source Computational Biology Framework in Python. *BMC Bioinformatics* **2018**, *19* (1), 346\. https://doi.org/10.1186/s12859-018-2367-z.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(37)	Wang, R.; Fang, X.; Lu, Y.; Wang, S. The PDBbind Database: Collection of Binding Affinities for Protein−Ligand Complexes with Known Three-Dimensional Structures. *J. Med. Chem.* **2004**, *47* (12), 2977–2980. https://doi.org/10.1021/jm030580l.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(38)	Buttenschoen, M.; Morris, G. M.; Deane, C. M. PoseBusters: AI-Based Docking Methods Fail to Generate Physically Valid Poses or Generalise to Novel Sequences. *Chem. Sci.* **2024**, *15* (9), 3130–3139. https://doi.org/10.1039/D3SC04185A.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(39)	Lin, Z.; Akin, H.; Rao, R.; Hie, B.; Zhu, Z.; Lu, W.; Smetanin, N.; Verkuil, R.; Kabeli, O.; Shmueli, Y.; Dos Santos Costa, A.; Fazel-Zarandi, M.; Sercu, T.; Candido, S.; Rives, A. Evolutionary-Scale Prediction of Atomic-Level Protein Structure with a Language Model. *Science* **2023**, *379* (6637), 1123–1130. https://doi.org/10.1126/science.ade2574.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(40)	Meli, R.; Biggin, P. C. Spyrmsd: Symmetry-Corrected RMSD Calculations in Python. *J Cheminform* **2020**, *12* (1), 49\. https://doi.org/10.1186/s13321-020-00455-2.](https://www.zotero.org/google-docs/?ZE0bAS)  
[(41)	McNutt, A. T.; Bisiriyu, F.; Song, S.; Vyas, A.; Hutchison, G. R.; Koes, D. R. Conformer Generation for Structure-Based Drug Design: How Many and How Good? *J. Chem. Inf. Model.* **2023**, *63* (21), 6598–6607. https://doi.org/10.1021/acs.jcim.3c01245.](https://www.zotero.org/google-docs/?ZE0bAS)

[image1]: docs/api/templ.png